### PR TITLE
[Paddle-TRT] Remove TensorRT deprecated API

### PR DESCRIPTION
**Pull request by zlsh80826**
### PR types

Others
### PR changes

Others
### Describe

Some usage of TensorRT is deprecated in TensorRT6, and those deprecation will be removed from TensorRT8. This PR intend to use the latest APIs which are compatible with TensorRT6-8.
1. DimsCHW -> Dims3 ([Reference](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-601/tensorrt-api/c_api/deprecated.html#_deprecated000001))
2. DimsNCHW -> Dims4 ([Reference](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-601/tensorrt-api/c_api/deprecated.html#_deprecated000002))
3. createNetwork is equal to createNetworkV2(0U) ([Reference](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-601/tensorrt-api/c_api/deprecated.html#_deprecated000017))
4. ``IBuilder::allowGPUFallback``, ``IBuilder::setDLACore, IBuilder::setFp16Mode``, ``IBuilder::setInt8Mode``, ``IBuilder::setMaxWorkspaceSize``, ``IBuilder::setStrictTypeConstraints`` are deprecated. Those configurations should be set by ``IBuilderConfig``. ([Reference](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-601/tensorrt-api/c_api/deprecated.html))
5. Remove duplicated int8/fp16 configurations for dynamic shape
6. ``buildCudaEngine`` is deprecated and it is replaced by ``buildEngineWithConfig`` (See 4.)
7. ``deserializeCudaEngine`` with 3 arguments is used for v1 plugins, and v1 plugins will be exclude from TensorRT8.

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/engine.cc [hunk 1]
```diff
@@ -212,30 +208,18 @@ void TensorRTEngine::FreezeNetwork() {
           Vec2TRT_Dims(optim_input_shape_[input.first], input.first, true));
     }
     infer_builder_config_->addOptimizationProfile(optim_profile_);
-    infer_builder_config_->setMaxWorkspaceSize(max_workspace_);
-    if (enable_int8) {
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/engine.cc:216]: 此处这个int8不生效的问题是否已经修复？

File: paddle/fluid/inference/tensorrt/engine.cc [hunk 2]
```diff
@@ -34,17 +34,16 @@ void TensorRTEngine::InitNetwork() {
...
-    infer_ptr<nvinfer1::IBuilderConfig> infer_builder_config_;
-    optim_profile_ = infer_builder_->createOptimizationProfile();
-#endif
   } else {
-    infer_network_.reset(infer_builder_->createNetwork());
+    infer_network_.reset(infer_builder_->createNetworkV2(0U));
   }
+
+  infer_builder_config_.reset(infer_builder_->createBuilderConfig());
+  infer_ptr<nvinfer1::IBuilderConfig> infer_builder_config_;
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/engine.cc]: 此处定义这个局部变量infer_builder_config_的作用是？是不是跟类的原有成员变量冲突了

Comment (zlsh80826) [reply paddle/fluid/inference/tensorrt/engine.cc]: 搬移位置時沒特別看, 現在看來 line 45 行特別奇怪, 應該是可以 remove 掉的

**Review by shangzhizhou — APPROVED** [review]
LGTM

**Review by cryoco — APPROVED** [review]
LGTM

**Pull request merged**